code/modules/mob/living/blood.dm

```diff
@@ -162,9 +162,26 @@
 						remove_status_effect(/datum/status_effect/debuff/bleeding)
 						remove_status_effect(/datum/status_effect/debuff/bleedingworse)
 
-			if(blood_volume <= BLOOD_VOLUME_BAD)
-				adjustOxyLoss(blood_volume <= BLOOD_VOLUME_SURVIVE ? 3 : 1)
-			else if((blood_volume > BLOOD_VOLUME_SURVIVE) || HAS_TRAIT(src, TRAIT_BLOODLOSS_IMMUNE))
+			// Automatic unconsciousness at BAD threshold
+			if(blood_volume <= BLOOD_VOLUME_BAD && !IsUnconscious())
+				Unconscious(10 SECONDS)
+				to_chat(src, span_userdanger("The world fades to black..."))
+
+			// CON-based soft crit threshold: higher CON = survive at lower blood levels
+			// Base is 122 at 10 CON, -10 blood per CON point above 10
+			// This means high CON characters enter soft crit later (at lower blood volume)
+			var/soft_crit_threshold = BLOOD_VOLUME_SURVIVE - ((STACON - 10) * 10)
+			soft_crit_threshold = clamp(soft_crit_threshold, 40, 200)
+
+			// Progressive oxygen damage based on blood loss severity
+			// Much lower rates for extended bleedout time (120-160 seconds)
+			if(blood_volume <= 40) // Death threshold - ~7% blood
+				adjustOxyLoss(2) // Rapid death
```

Comment:
pre-build the oxyloss value amount in a distinct variable and apply it with 1 ``adjustOxyLoss`` call where possible to minimize health updates, big contributors to lag